DAOS-15984 test: Detect server pid upon startup

phender opened the pull request:
Collect the server engine pid when detecting server start.

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: HarnessSetupVmTest

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

phender committed:
DAOS-15984 test: Detect server pid upon startup

Collect the server engine pid when detecting server start.

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: HarnessSetupVmTest

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

daosbuild1 commented:
Test stage Functional on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14524/1/execution/node/823/log

phender committed:
Fix dictionary key typo

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: HarnessSetupVmTest

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

phender committed:
Added kill_rank() method to DaosServerManager.

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: HarnessSetupVmTest

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

phender committed:
Added information message for pid mapping update.

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: HarnessSetupVmTest

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

daosbuild1 commented:
Test stage Functional on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14524/4/execution/node/799/log

phender committed:
Fixes.

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: HarnessSetupVmTest

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

phender committed:
Updating ServerRankFailure tests with kill rank method.

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: pr ServerRankFailure RebuildMdtest
Skip-func-hw-test-medium-verbs-provider: true

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

daosbuild1 commented:
Test stage Python Bandit check completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14524/6/execution/node/145/log

phender committed:
Support multiple random ranks; killing multiple pids

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: pr ServerRankFailure RebuildMdtest test_sub_process_command
Skip-func-hw-test-medium-verbs-provider: true

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

phender committed:
Fix edit.

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: pr ServerRankFailure RebuildMdtest test_sub_process_command

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

daosbuild1 commented:
Test stage Functional on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14524/8/execution/node/799/log

phender committed:
Merge branch 'master' into pahender/DAOS-15984

phender committed:
Applying fixes.

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: pr ServerRankFailure RebuildMdtest test_sub_process_command

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

daosbuild1 commented:
Test stage Functional on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14524/9/execution/node/798/log

phender committed:
Merge branch 'master' into pahender/DAOS-15984

phender committed:
Fix cart_self_test.py

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: pr ServerRankFailure RebuildMdtest test_sub_process_command

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

daosbuild1 commented:
Test stage Functional on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14524/11/execution/node/823/log

phender committed:
Fix logic.

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: pr ServerRankFailure RebuildMdtest test_sub_process_command

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14524/12/execution/node/1124/log

phender committed:
Merge branch 'master' into pahender/DAOS-15984

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: pr ServerRankFailure RebuildMdtest test_sub_process_command

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14524/13/execution/node/1099/log

phender committed:
Merge branch 'master' into pahender/DAOS-15984

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: pr ServerRankFailure RebuildMdtest test_sub_process_command

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14524/14/execution/node/1040/log
